autopilot/contractor.go

```diff
@@ -40,16 +52,75 @@ func newContractor(ap *Autopilot) *contractor {
 
 func (c *contractor) applyConsensusState(cfg Config, state bus.ConsensusState) {
 	c.blockHeight = state.BlockHeight
-	// TODO: update current period
+	if c.blockHeight > c.currentPeriod+cfg.Contracts.Period {
+		c.currentPeriod += cfg.Contracts.Period
```

Comment:
One thing I noticed while doing some testing was the fact that `c.currentPeriod` isn't initialised ever. I think the first time you start the contractor it should block until consensus is synced before a user can enable contract formations.
When contract formations are enabled, we initialise the currentPeriod to the current block height. That way you don't end up forming a contract that is midway through the period after forming.
It also means the renew estimate is more accurate because there is more time to collect spending metrics.

Reply:
Do users have to explicitly enable contract formations in the new renter? Is it like `siad` where you have to set the allowance? Do you want to implement this blocking mechanism now? I just set the current period to the tip of the chain now. The autopilot already does nothing until we're synced to `applyConsensusState` would set the current period to the tip and then go on to form contracts.